src/System.Net.Http/tests/FunctionalTests/CancellationTest.cs

```diff
@@ -15,110 +15,108 @@ namespace System.Net.Http.Functional.Tests
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task GetAsync_ResponseContentRead_CancelUsingTimeoutOrToken_TaskCanceledQuickly(bool useTimeout)
         {
-            var stopwatch = new Stopwatch();
-            stopwatch.Start();
+            var cts = new CancellationTokenSource(); // ignored if useTimeout==true
+            TimeSpan timeout = useTimeout ? new TimeSpan(0, 0, 1) : Timeout.InfiniteTimeSpan;
+            CancellationToken cancellationToken = useTimeout ? CancellationToken.None : cts.Token;
 
-            using (var client = new HttpClient())
+            using (var client = new HttpClient() { Timeout = timeout })
             {
-                client.Timeout = new TimeSpan(0, 0, 1);
-
-                Task<HttpResponseMessage> getResponse =
-                    client.GetAsync(s_fastHeadersSlowBodyServer, HttpCompletionOption.ResponseContentRead);
-
-                stopwatch.Restart();
-                await Assert.ThrowsAsync<TaskCanceledException>(
-                    () => getResponse);
-                stopwatch.Stop();
-                _output.WriteLine("GetAsync() completed at: {0}", stopwatch.Elapsed.ToString());
-                Assert.True(stopwatch.Elapsed < new TimeSpan(0,0,3), "Elapsed time should be short");
-            }
-        }
-
-        [ActiveIssue(8663)]
-        [Fact]
-        public async Task GetIncludesReadingResponseBody_CancelUsingToken_TaskCanceledQuickly()
-        {
-            var stopwatch = new Stopwatch();
-            stopwatch.Start();
+                var triggerResponseWrite = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+                var triggerRequestCancel = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
 
-            var cts = new CancellationTokenSource();
-            using (var client = new HttpClient())
-            {
-                Task<HttpResponseMessage> getResponse =
-                    client.GetAsync(s_fastHeadersSlowBodyServer, HttpCompletionOption.ResponseContentRead, cts.Token);
-
-                Task ignore = Task.Delay(new TimeSpan(0, 0, 1)).ContinueWith(_ =>
+                await LoopbackServer.CreateServerAsync(async (server, url) =>
                 {
-                    _output.WriteLine("Calling cts.Cancel() at: {0}", stopwatch.Elapsed.ToString());
-                    cts.Cancel();
+                    Task serverTask = LoopbackServer.AcceptSocketAsync(server, async (socket, stream, reader, writer) =>
+                    {
+                        while (!string.IsNullOrEmpty(await reader.ReadLineAsync())) ;
+                        await writer.WriteAsync(
+                            "HTTP/1.1 200 OK\r\n" +
+                            $"Date: {DateTimeOffset.UtcNow:R}\r\n" +
+                            "Content-Length: 16000\r\n" +
+                            "\r\n" +
+                            "less than 16000 bytes");
```

Comment:
Switching to a loopback server was a TODO that I had.  But this change is more than just using a loopback server. The intent is the tests was to have response headers sent right away but then have response body payload arrive later ("FastHeadersSlowBody").

This change using the LoopbackServer is fundamentally different.  It sends the response headers and 99% of the payload out right away. It's "99%" because the test code is intentionally sending out less bytes than the 'Content-Length'.  While this is an interesting test, it does not exercise the same codepaths in WinHttpHandler, for example, that I was explicitly testing against. I.e. I was testing for the WinHttpQueryDataAvailable() call taking a long time because there were 0 bytes available.  So, this change here makes this a totally different kind of test.